### PR TITLE
Feature/layout builder

### PR DESCRIPTION
**Pull request by peterekepeter**
Implementation of the layout generator,
This enables a lot more flexibility and customizations.

### CODE REVIEW

**Comment by peterekepeter** [comment]
I'd say this is ready to be merged.

**Comment by gazlaws-dev** [comment]
This is a wonderful effort, thank you for the great framework to build on! Sorry for the long delay, life got in the way. Code review passed.

**Comment by gazlaws-dev** [comment]
Fixes #49

**Comment by a121345** [comment]
```
[]()
```

File: .gitignore
Comment (a121345) [inline .gitignore:1]: https://github.com/gazlaws-dev/codeboard/commit/be4f0fd40779c7064fca0a03a948d8cc8af91128

**Pull request merged**